src/hooks/useWindowSize.ts

```diff
@@ -0,0 +1,50 @@
+import { useState, useEffect } from 'react';
+
+import { isServer } from 'utils';
+
+type TWindowSize = {
+  width: number | undefined,
+  height: number | undefined,
+};
+
+const useWindowSize = ():TWindowSize => {
+  const [ windowSize, setWindowSize ] = useState<TWindowSize> ( ():TWindowSize => {
```

Comment:
А нам точно нужна именно такая логика с callback? В смысле как-то выглядит оно как излишнее усложнение...

Reply:
А как бы ту сделала? Я наверное не совсем понимаю о чем ты... (

Reply:
❗️
Это можно запихать в useEffect, там будет органичнее

Reply:
Не получается. Мне нужно зполнить windowSize еще до вызова useEffect.

Reply:
❗️
Не очень понимаю зачем, но ладно, давай тогда просто код-стайл причешем и ок
```suggestion
  const [ windowSize, setWindowSize ] = useState<TWindowSize>(():TWindowSize => {
```